### ISSUE TITLE
Call of overloaded 'add(JsonArraySubscript)' is ambiguous

### ISSUE THREAD

**Issue by mozgy** [bblanchon/ArduinoJson#122]
Lets say that

```c++
JsonObject& data = { "foo":1, "nested": [ { "bar":2 } { "bar":3 } ... ] };
JsonArray& newdata = json.createArray();
JsonArray& tmp = data["nested"];
```

please add that following works

```c++
newdata.add( tmp[0] );
```

after #120 fix it's possible to do that with

```
String strTmp; tmp[0].printTo( strTmp ); newdata.add( strTmp );
```

but it looks a bit overkill using String buffer copy for that

TIA,

**Comment by bblanchon** [comment]
I think it should already work.
Maybe I don't understand.
Can you post a program that demonstrate the issue?

**Comment by mozgy** [comment]
For starters it won't even compile ..
call of overloaded 'add(ArduinoJson::JsonArraySubscript)' is ambiguous

**Comment by bblanchon** [comment]
Fix in [version 5.0.4](https://github.com/bblanchon/ArduinoJson/releases/tag/v5.0.4)

**Comment by mozgy** [comment]
ty
